From dec8ed438e2906dc8482747a0744ce588998608d Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 11 May 2023 12:02:54 +0200
Subject: [PATCH 0394/1695] drivers: stm32_rng: fix comment in
 stm32_rng_pm_resume()

Fix comment about the application of the RNG configuration in
stm32_rng_pm_resume(). Old comment mentioned reserved bits.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 15cb24838c9994430333d23f633fd7ac76a192a0 drivers: stm32_rng: fix comment in stm32_rng_pm_resume()]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I8a39ef1793a62a99f876413fbade810508de8600
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/305797
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_rng.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/core/drivers/stm32_rng.c b/core/drivers/stm32_rng.c
index 68c6f116e..c9a4f7914 100644
--- a/core/drivers/stm32_rng.c
+++ b/core/drivers/stm32_rng.c
@@ -438,9 +438,10 @@ static TEE_Result stm32_rng_pm_resume(void)
 
 	if (stm32_rng->ddata->has_cond_reset) {
 		/*
-		 * Correct configuration in bits [29:4] must be set in the same
-		 * access that set RNG_CR_CONDRST bit. Else config setting is
-		 * not taken into account.
+		 * Configuration must be set in the same access that sets
+		 * RNG_CR_CONDRST bit. Otherwise, the configuration setting is
+		 * not taken into account. CONFIGLOCK bit is always cleared in
+		 * this configuration.
 		 */
 		io_write32(base + RNG_CR, stm32_rng->pm_cr | RNG_CR_CONDRST);
 
-- 
2.47.2

